src/main/java/seedu/java/util/TaskList.java

```diff
@@ -0,0 +1,134 @@
+package seedu.java.util;
+
+import java.util.ArrayList;
+
+/**
+ * Handles Task and their operations.
+ *
+ * ArrayList<Task> getTaskArr()
+ * String listToPrint()
+ * String read()
+ */
+public class TaskList {
+    private ArrayList<Task> taskArr;
+    private String path = "data/duke.txt";
+
+    public TaskList() {
+        this.taskArr = new ArrayList<>();
+    }
+
+    public TaskList(ArrayList<Task> taskArr) {
+        this.taskArr = taskArr;
+    }
+
+    /**
+     * Returns its ArrayList of Tasks
+     * @return taskArr
+     */
+    public ArrayList<Task> getTaskArr() {
+        return taskArr;
+    }
+
+    private String totalTask() {
+        return "You have " + (taskArr.size()) + " task on your list.";
+    }
+
+    private String addTodo(String task) {
+        taskArr.add(new Todo(task));
+        return "Affirmative. Adding a to-do task :) \n" + "added: " + task + "\n" + totalTask();
+    }
+
+    private String addDead(String task, String timing) {
+        taskArr.add(new Deadline(task, timing));
+        return "Affirmative. Adding a task with Deadline :) \n" + "added: " + task + "\n" + totalTask();
+    }
+
+    private String addEvent(String task, String timing) {
+        taskArr.add(new Event(task, timing));
+        return "Affirmative. Adding an event :) \n" + "added: " + task + "\n" + totalTask();
+    }
+
+    private String deleteTask(int index) {
+        taskArr.remove(index - 1);
+        return "Deleting this task. now you have less things to do :)" + "\n" + totalTask();
+    }
+
+    /**
+     * Completes a task and returns a string to be printed by the UI.
+     * @param index - the task that you want printed
+     * @return String to print
+     */
+    private String doneTask(int index) {
+        taskArr.set(index - 1, taskArr.get(index - 1).completeTask());
+        return "Complete task! :)" + "\n" + totalTask();
+    }
+
+    /**
+     * Returns a String of list of Task.
+     * @return String of list
+     */
+    public String listToPrint() {
+        int i = 1;
+        String toPrint = "";
+        for (Task x : taskArr) {
+            toPrint += (i++) + "" + x.toString() + "\n";
+        }
+        return toPrint;
+    }
+
+    private String listToPrint(ArrayList<Task> tasks) {
```

Comment:
Seems like unnecessary code duplication! (the two `listToPrint` methods)

Would it be a better idea to just use the one with signature `listToPrint(ArrayList<Task> tasks)` and pass in `this.taskArr` when required? It may also make it more evident which list you are printing